zerver/views/registration.py

```diff
@@ -76,8 +76,12 @@ def check_prereg_key_and_redirect(request: HttpRequest, confirmation_key: str) -
 
 @require_post
 def accounts_register(request: HttpRequest) -> HttpResponse:
-    key = request.POST['key']
-    confirmation = Confirmation.objects.get(confirmation_key=key)
+    try:
+        key = request.POST['key']
```

Comment:
Won't we get a `KeyError` if the user doesn't POST anything? We should catch that too I guess

Reply:
hmm ok. Should I do something like `request.POST.get('key', '')` rather than another try catch?

Reply:
I guess that should work